components/Layout.tsx

```diff
@@ -1,50 +1,83 @@
+// react imports
 import React, { ReactNode } from "react";
+
+// next imports
+import Head from 'next/head';
+import {useRouter} from 'next/router'
+
+// component imports
 import Header from "@/components/Header";
 
+//local imports
+import { PAGES } from "src/utils/constants";
+
 type Props = {
   children: ReactNode;
 };
 
-const Layout: React.FC<Props> = (props) => (
-  <div>
-    <Header />
-    <div className="layout">{props.children}</div>
-    <style jsx global>{`
-      html {
-        box-sizing: border-box;
-      }
-
-      *,
-      *:before,
-      *:after {
-        box-sizing: inherit;
-      }
-
-      body {
-        margin: 0;
-        padding: 0;
-        font-size: 16px;
-        font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
-          Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji",
-          "Segoe UI Symbol";
-        background: rgba(0, 0, 0, 0.05);
-      }
-
-      input,
-      textarea {
-        font-size: 16px;
-      }
-
-      button {
-        cursor: pointer;
-      }
-    `}</style>
-    <style jsx>{`
-      .layout {
-        padding: 0 2rem;
-      }
-    `}</style>
-  </div>
-);
+const Layout: React.FC<Props> = (props) => {
+
+  const router = useRouter();
+
+  const filterFunc = (item) =>{
+    return item[1].link === router.pathname
+  }
+  
+  const currentPage = () => {
+    if(router.pathname === '/') return 'Home'
+    const entry = Object.entries(PAGES).filter(filterFunc)
```

Comment:
There's an easier way to do this, we don't need to convert it to an array and can do:
```
return PAGES.find((page) => page?.name === router.pathname)?.name;
```
It's also good practice to use dot notation and using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) as opposed to square brackets; the optional chaining prevents any errors from getting thrown and is a nice safe guard!

Reply:
I'd actually like to discuss Optional Chaining as a code standard (maybe also on Monday) cuz it was a huge debate at my last company lol. 

Our team over there was on the side of optional chaining not showing errors is overall more of a negative than a positive as it can unintentionally hide issues early, and thus causing longer troubleshooting times and less safe code overall, hence my lack of use of it here. The gist of the thought process was if you can't guarantee it exists, you _must_ have countermeasures for when it doesn't, which also prevents future breakages for users. 

That being said, in this scenario I can see using it in the function and just protecting against the return (i.e. if undefined assign to Home), but since it is such a hot debate I didn't want to use it until it was discussed. We were also designing for public use, but since this is internal I'm not sure if errors being shown to the user is as much of a concern as it was for public-facing ecommerce websites.

Either way I can comment out the current code and write up the optional chaining version for now, I just wanted to open the discussion since it's relevant!

Reply:
Also side note regardless of code standard this:
` const currentPage = () => PAGES.find((page) => page.link === router.pathname)?.name || "Home" `
looks a ton cleaner lol

Reply:
> I'd actually like to discuss Optional Chaining as a code standard (maybe also on Monday) cuz it was a huge debate at my last company lol.
> 
> Our team over there was on the side of optional chaining not showing errors is overall more of a negative than a positive as it can unintentionally hide issues early, and thus causing longer troubleshooting times and less safe code overall, hence my lack of use of it here. The gist of the thought process was if you can't guarantee it exists, you _must_ have countermeasures for when it doesn't, which also prevents future breakages for users.
> 
> That being said, in this scenario I can see using it in the function and just protecting against the return (i.e. if undefined assign to Home), but since it is such a hot debate I didn't want to use it until it was discussed. We were also designing for public use, but since this is internal I'm not sure if errors being shown to the user is as much of a concern as it was for public-facing ecommerce websites.
> 
> Either way I can comment out the current code and write up the optional chaining version for now, I just wanted to open the discussion since it's relevant!

Yes that's an excellent point! Yes, we should theoretically not need to use optional chaining if a) our API contracts are clear and defined; and b) we have safeguards in place on the backend. So in a vacuum yes, I agree with you! But given the nature of this project (admin facing only) and that we have a lot of people in the learning phase I think it's fine to push for optional chaining - mostly because I don't think we're at a point where we can guarantee what data exists 😆 . We would rather have a console log error than an runtime error, since runtime errors will typically prevent components from rendering; and buildtime SSR errors prevent the page from loading